test/protocols/binary_test.exs

```diff
@@ -1,80 +1,183 @@
 defmodule BinaryProtocolTest do
+  use ThriftTestCase, gen_erl: true
   use ExUnit.Case
 
-  @thrift_file_path "./test/fixtures/app/thrift/simple.thrift"
+  @moduletag :integration
 
-  import ParserUtils
+  def round_trip_struct(data, serializer_mf, deserializer_mf) do
+    {serializer_mod, serializer_fn} = serializer_mf
+    {deserializer_mod, deserializer_fn} = deserializer_mf
 
-  setup_all do
-    @thrift_file_path
-    |> parse_thrift
-    |> compile_module
+    serialized = :erlang.apply(serializer_mod, serializer_fn, [:struct, data])
```

Comment:
It's so you can serialize arbitrary things. 
```elixir
serialize(:struct, %MyStruct{})
serialize({:list, :struct}, [%MyStruct])
serialize({:list, i32}, [1, 2, 3, 4])
...etc
```
We can add the serialize/1, but this isn't germane to this PR.